fact_extractor/plugins/unpacking/generic_carver/code/generic_carver.py

```diff
@@ -18,9 +20,22 @@ def unpack_function(file_path, tmp_dir):
 
     logging.debug('File Type unknown: execute binwalk on {}'.format(file_path))
     output = execute_shell_command('binwalk --extract --carve --signature --directory  {} {}'.format(tmp_dir, file_path))
+
+    drop_underscore_directory(tmp_dir)
+
     return {'output': output}
 
 
+def drop_underscore_directory(tmp_dir):
+    extracted_contents = list(Path(tmp_dir).iterdir())
+    if not extracted_contents:
+        return
+    if not len(extracted_contents) == 1 or not extracted_contents[0].name.startswith('_'):
+        return
+    for result in extracted_contents[0].iterdir():
+        shutil.move(str(result), str(result.parent.parent))
```

Comment:
you do not remove the underscore folder, because FACT ignores empty folders. However, if you do a "tar repack" or use the fact_extractor on its own the folder is still there, isn't it?